server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/GetDiscoveredNodesRequest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.action.admin.cluster.bootstrap;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.io.IOException;
+
+/**
+ * Request the set of master-eligible nodes discovered by this node. Most useful in a brand-new cluster as a precursor to setting the
+ * initial configuration using {@link BootstrapClusterRequest}.
+ */
+public class GetDiscoveredNodesRequest extends ActionRequest {
+
+    private int minimumNodeCount = 1;
```

Comment:
I changed this name in bdbd600510b488de2629f2e41bdbc194560e5acb after updating the docs and realising that setting `waitForNodes` > 1 doesn't actually cause a wait, it just yields an error. But this may be a consequence of the default timeout being zero and/or the extra validation that you asked to remove.

Reply:
Renamed it back in d0a62cb15d60e5f56cc40e791d939e14ae259bf2.